### ISSUE TITLE
Option to preserve the line "Created by Alice on 12/08/2016." in file header?

### ISSUE THREAD

**Issue by honghaoz** [nicklockwood/SwiftFormat#776]
I've been longing for a feature to preserve the line `Created by Alice on 12/08/2016` in the original header.

It is a common template in Xcode that a file has a header like:
```
//
//  Arguments.swift
//  SwiftFormat
//
//  Created by Nick Lockwood on 07/08/2018.
//  Copyright © 2018 Nick Lockwood.
//
```

I would like to have `swiftformat` to support a header option like `--header "//\n// {file}\n//\n// {created_by}\n// Copyright © {year} Example. All rights reserved.\n//"`

So it can generate a header like:
```
//
//  Arguments.swift
//
//  Created by Nick Lockwood on 07/08/2018.
//  Copyright © 2020 Example. All rights reserved.
//
```

The purpose is to keep the original author's name and the original creation date untouched.

One reason is the current `{created}` is not accurate; it depends on the locale and timezone of the machine running the script, which can have an unexpected date in different machines like CIs.

Another reason is there's no way to keep the author's name with the current version.

Basically, the idea is to replace the `{created_by}` placeholder with the content of `Created by ...` in the original header if it appears.

Thanks!

**Comment by honghaoz** [comment]
I made a quick implementation here: https://github.com/chouti-dev/SwiftFormat/pull/1/files

**Comment by markeissler** [comment]
I just started using swiftformat for a large project. The problem I'm seeing with `{created}` is that it is referencing the file creation date on the host that it is running on. In this case, it's the date that I happened to clone the project.

When creating the file initially this is perfectly fine. But in later revisions of the file, the header is rewritten (which is good) but the created line is also updated. What this means is that the created line will change every time the file is modified which is not the way it should work. Correct me if I'm wrong.

There needs to be a way to not alter certain lines from the template if they already exist.

**Comment by honghaoz** [comment]
@markeissler this change (https://github.com/chouti-dev/SwiftFormat/pull/1/files) is what I have been using in my forked version. Lines prefixes with `Created by` will be untouched.

**Comment by markeissler** [comment]
@honghaoz Yes. I just realized that we are looking for the same feature. After looking at your PR. I should have re-read your description above more slowly!

Did you open a PR here on this repo?

**Comment by honghaoz** [comment]
@markeissler  I haven't made a PR to this repo yet cause I thought it was a quick hack. Surprisingly it worked well for me for more than a year. 
I can try to make a PR.

**Comment by honghaoz** [comment]
@markeissler sounds like a more smart strategy. But I'm not sure what's the use case of the extracted date. If I understand correctly, seems like you can't use the extracted date freely in other positions.
For example, for the header line
```swift
//  Created by Bob Smith on 02/14/2022.
```
Only the `--header "//  Created by Bob Smith on {created}."` works expected on multiple format runs.

If you ever changes the header option to `--header "//  Created on {created} by Bob Smith."`, you have a problem of the regex pattern won't match, thus it can't extra the date. Then you use the fallback behavior which will may ruin the original created date.

Not sure if I missed anything. 

Regarding the PR, I feel my solution is still not general enough to be in the tool, as my implementation only works for lines prefixed exactly with "//  Created by".

**Comment by markeissler** [comment]
@honghaoz You are 100% correct. You cannot change the header format that contains `{created}` once you've created it at least not automatically. That would have to be a manual process like the one I just went through when I applied `swiftformat` to all of the files in a large project for the first time.

The reason the date is parsed is two-fold:

1. Extracting the date let's us just continue with the normal process of outputting the entire header section. The `date` value is simply set to the extracted value instead of the current date.
2. This method supports languages. If we were to hard code `Created on` and search for that (or some other fixed hard-coded string) it means bypassing the value of `--header` all together and then only supporting an english header line.

Yes, there is the limitation of not being able to change this part of the header (whichever line contains `{created}`) automatically but in use it's unlikely that the format would change very often if ever.

I think the project would be over-engineered if it supported the limited use case of providing a feature to specify both an existing header line with a placeholder, and then a new header line with the same placeholder. The typical use case (I think) is that once a `{created}` or `{created.year}` header has been written, you don't want that value to be overridden with each subsequent application of `swiftformat` on the file. Because then it would actually be an `{updated}` field instead. ¯\\_(ツ)_/¯
